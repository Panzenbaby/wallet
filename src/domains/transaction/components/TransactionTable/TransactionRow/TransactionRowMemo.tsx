import { Icon } from "app/components/Icon";
import { Tooltip } from "app/components/Tooltip";
import React from "react";

interface Properties {
	memo?: string;
}

export const TransactionRowMemo = ({ memo }: Properties) => (
	<div
		data-testid="TransactionRowMemo"
		className="inline-flex space-x-1 align-middle text-theme-secondary-700 dark:text-theme-secondary-600"
	>
		{!!memo && (
			<Tooltip className="break-all" content={memo}>
				<span className="p-1">
					<Icon data-testid="TransactionRowMemo__vendorField" name="FileLines" size="lg" />
				</span>
			</Tooltip>
		)}
	</div>
);
