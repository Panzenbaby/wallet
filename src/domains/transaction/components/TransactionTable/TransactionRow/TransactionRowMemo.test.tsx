import { render, screen } from "@testing-library/react";
import React from "react";

import { TransactionRowMemo } from "./TransactionRowMemo";

describe("TransactionRowMemo", () => {
	it("should show memo", () => {
		const { rerender, asFragment } = render(<TransactionRowMemo memo="memo" />);

		expect(screen.getByTestId("TransactionRowMemo__vendorField")).toBeInTheDocument();

		rerender(<TransactionRowMemo memo={undefined} />);

		expect(screen.queryByTestId("TransactionRowMemo__vendorField")).not.toBeInTheDocument();

		expect(asFragment()).toMatchSnapshot();
	});
});
