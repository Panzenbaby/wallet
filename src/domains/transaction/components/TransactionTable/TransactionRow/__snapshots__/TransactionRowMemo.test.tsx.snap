// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionRowMemo should show memo 1`] = `
<DocumentFragment>
  <div
    class="inline-flex space-x-1 align-middle text-theme-secondary-700 dark:text-theme-secondary-600"
    data-testid="TransactionRowMemo"
  />
</DocumentFragment>
`;
